openpype/hosts/houdini/api/lib.py

```diff
@@ -559,29 +559,51 @@ def get_template_from_value(key, value):
     return parm
 
 
-def get_frame_data(node):
+def get_frame_data(node, asset_data=None, log=None):
     """Get the frame data: start frame, end frame and steps.
 
     Args:
         node(hou.Node)
 
     Returns:
-        dict: frame data for star, end and steps.
+        dict: frame data for start, end and steps.
 
     """
+    if asset_data is None:
+        asset_data = {}
+
+    if log is None:
+        log = self.log
```

Comment:
Where is `self` defined?

Reply:
Here: https://github.com/ynput/OpenPype/blob/520ff86cd044c15328fa936cc452301836626943/openpype/hosts/houdini/api/lib.py#L32

Which refers to the module's global namespace - and thus when `log` gets set just below suddenly `self.log` exists ;)

Reply:
At the top of `lib.py` 
also, `self.log` is used in many places in the same file

https://github.com/ynput/OpenPype/blob/b1dbbcd954e242b7d2357b7d49d8bd610db5278d/openpype/hosts/houdini/api/lib.py#L32